src/ec2_metadata/__init__.py

```diff
@@ -184,6 +184,19 @@ def reservation_id(self) -> str:
     def security_groups(self) -> list[str]:
         return self._get_url(f"{self.metadata_url}security-groups").text.splitlines()
 
+    @cached_property
+    def tags(self) -> dict[str, str]:
+        tags = {}
+
+        resp = self._get_url(f"{self.metadata_url}tags/instance/", allow_404=True)
+        if not resp.status_code == 404:
+            tags_keys = [line.rstrip("/") for line in resp.text.splitlines()]
+            for tag_key in tags_keys:
+                resp = self._get_url(f"{self.metadata_url}tags/instance/{tag_key}")
+                tag_value = resp.text
+                tags[tag_key] = tag_value
+        return tags
```

Comment:
Due to [query throttling](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instancedata-data-retrieval.html#instancedata-throttling) I think we should lazily fetch tag values. Otherwise fetching all of them when the consumign code needs just one is expensive.

Thus we should return a special dict subclass here.